From b3123503319d32f7fa476c9b3a3a105a2b240ae0 Mon Sep 17 00:00:00 2001
From: Felipe Balbi <balbi@ti.com>
Date: Mon, 14 Feb 2011 19:30:41 +0200
Subject: [PATCH 0298/1092] arm: omap: pass Retu ID via platform_data

Instead of hardcoding on the driver, pass
that via platform_data instead.

Signed-off-by: Felipe Balbi <balbi@ti.com>
---
 arch/arm/mach-omap1/board-nokia770.c   | 1 +
 arch/arm/mach-omap2/board-n8x0.c       | 1 +
 arch/arm/plat-omap/include/plat/cbus.h | 1 +
 3 files changed, 3 insertions(+)

diff --git a/arch/arm/mach-omap1/board-nokia770.c b/arch/arm/mach-omap1/board-nokia770.c
index c9c8e62..1ba0ffd 100644
--- a/arch/arm/mach-omap1/board-nokia770.c
+++ b/arch/arm/mach-omap1/board-nokia770.c
@@ -110,6 +110,7 @@ static struct resource retu_resource[] = {
 static struct cbus_retu_platform_data nokia770_retu_data = {
 	.irq_base	= CBUS_RETU_IRQ_BASE,
 	.irq_end	= CBUS_RETU_IRQ_END,
+	.devid		= CBUS_RETU_DEVICE_ID,
 };
 
 static struct platform_device retu_device = {
diff --git a/arch/arm/mach-omap2/board-n8x0.c b/arch/arm/mach-omap2/board-n8x0.c
index e39c359..28f9a60 100644
--- a/arch/arm/mach-omap2/board-n8x0.c
+++ b/arch/arm/mach-omap2/board-n8x0.c
@@ -221,6 +221,7 @@ static struct resource retu_resource[] = {
 static struct cbus_retu_platform_data n8x0_retu_data = {
 	.irq_base	= CBUS_RETU_IRQ_BASE,
 	.irq_end	= CBUS_RETU_IRQ_END,
+	.devid		= CBUS_RETU_DEVICE_ID,
 };
 
 static struct platform_device retu_device = {
diff --git a/arch/arm/plat-omap/include/plat/cbus.h b/arch/arm/plat-omap/include/plat/cbus.h
index e7ace57..563b1c2 100644
--- a/arch/arm/plat-omap/include/plat/cbus.h
+++ b/arch/arm/plat-omap/include/plat/cbus.h
@@ -34,6 +34,7 @@ struct cbus_host_platform_data {
 struct cbus_retu_platform_data {
 	int	irq_base;
 	int	irq_end;
+	int	devid;
 };
 
 #endif /* __PLAT_CBUS_H */
-- 
1.7.11.2

